File: app/Global.java
import models.IPAddress;
import play.Application;
import play.GlobalSettings;
import play.Logger;
import play.api.mvc.EssentialFilter;
import play.libs.F;
import play.mvc.Action;
import play.mvc.Http;
import play.filters.gzip.GzipFilter;
import play.mvc.SimpleResult;

import java.lang.reflect.Method;

public class Global extends GlobalSettings {

    public <T extends EssentialFilter> Class<T>[] filters() {
        return new Class[]{GzipFilter.class};
    }

    @Override
    public Action onRequest(Http.Request request, Method method) {
        String ip = request.remoteAddress();
        IPAddress ipAddress = IPAddress.findByIP(ip);
        if (ipAddress == null) {
            ipAddress = new IPAddress(ip);
            ipAddress.save();
        } else {
            if (ipAddress.isBanned()) {
                return new Action.Simple() {
                    @Override
                    public F.Promise<SimpleResult> call(Http.Context ctx) throws Throwable {
                        return F.Promise.pure((SimpleResult) status(0));
                    }
                };
            }
        }
        //ipAddress.login();
        return super.onRequest(request, method);
    }

    @Override
    public F.Promise<SimpleResult> onHandlerNotFound(Http.RequestHeader requestHeader) {
        return super.onHandlerNotFound(requestHeader);
    }

    public void onStart(Application app) {
        Logger.info("Application started");
    }

    public void onStop(Application app) {
        Logger.info("Application stopped");
    }


}

File: app/Global.scala

import com.antigenomics.vdjtools.misc.Software
import models.auth.User
import models.file.{IntersectionFile, ServerFile}
import org.apache.commons.io.FilenameUtils
import org.mindrot.jbcrypt.BCrypt
import play.api._
import play.api.mvc._
import play.api.mvc.Results._
import play.api.libs.concurrent.Akka
import server.ServerLogger

import scala.concurrent.duration._
import play.api.libs.concurrent.Execution.Implicits._
import securesocial.core._
import server.Configuration
import server.database.GlobalDatabase
import service.AuthService
import utils.CommonUtils

import scala.collection.JavaConversions._
import scala.concurrent.Future

/**
  * Created by bvdmitri on 20.03.16.
  */

object Global extends GlobalSettings {
  override def onStart(app: Application): Unit = {
    ServerLogger.info("Application has started : trying to update the database")
    GlobalDatabase.initDatabase()
    if (Configuration.automaticDatabaseUpdate) {
      updateDaemon(app)
    }

    if (Configuration.createDemoAccount) {
      if (!User.isUserExistByUUID("demo")) {
        val userId = "demo"
        val userService: UserService = new AuthService(app)
        val socialUser = new SocialUser(new IdentityId(userId, "userpass"),
          userId, userId, String.format("%s %s", userId, userId),
          Option.apply(userId), null, AuthenticationMethod.UserPassword,
          null, null, Some.apply(new PasswordInfo("bcrypt", BCrypt.hashpw(userId, BCrypt.gensalt()), null))
        )
        userService.save(socialUser)
      }
      val user = User.findByUUID("demo")
      for (file <- user.getFiles.toList) {
        ServerFile.deleteFile(file)
      }
      for (demoFile <- CommonUtils.getListOfFiles(Configuration.demoDatasetPath)) {

        FilenameUtils.getExtension(demoFile.getName) match {
          case "gz" =>
            val uniqueName = CommonUtils.randomAlphaNumericString(5)
            val destDir = user.getDirectoryPath + "/" + uniqueName
            CommonUtils.createDir(destDir)
            val dest = destDir + "/" + demoFile.getName
            CommonUtils.copyFile(demoFile.getAbsolutePath, dest)
            val intersectionFile: IntersectionFile = new IntersectionFile(user, demoFile.getName,
              uniqueName, destDir, dest, Software.VDJtools)
            intersectionFile.save()
          case _ =>
        }
      }
    }
  }

  override def onStop(app: Application): Unit = {
    ServerLogger.info("Application has stopped")
  }


  override def onHandlerNotFound(request: RequestHeader): Future[SimpleResult] = {
    Future(NotFound(views.html.common.notFound()))
  }

  def updateDaemon(app: Application): Unit = {
    Akka.system(app).scheduler.schedule(30 minutes, 1 day, new Runnable {
      override def run(): Unit = {
        ServerLogger.info("Trying to update the database")
        GlobalDatabase.update()
      }
    })
  }

}
